linker/Mono.Linker.Steps/MarkStep.cs

```diff
@@ -610,8 +619,58 @@ protected virtual void DoAdditionalTypeProcessing (TypeDefinition method)
 		{
 		}
 
-		void MarkTypeSpecialCustomAttributes (TypeDefinition type)
+		void MarkAssemblySpecialCustomAttributes(AssemblyDefinition assembly)
+		{
+			if (!assembly.HasCustomAttributes)
+				return;
+
+			foreach (CustomAttribute attribute in assembly.CustomAttributes) {
+				string attributeFullName = attribute.Constructor.DeclaringType.FullName;
+				switch (attributeFullName) {
+					case "System.Diagnostics.DebuggerDisplayAttribute":
+					case "System.Diagnostics.DebuggerTypeProxyAttribute":
+						if (_context.KeepMembersForDebuggerAttributes) {
```

Comment:
Extract this to special method so you don't need to do secondary attributeFullName check. Something like
`StoreDebuggerTypeTarget (_assemblyDebuggerDisplayAttributes, attribute);`